### PR TITLE
EUI-8303 Query Management orchestrator to get the mock data

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-8303

### Change description ###
Orchestrator to get the 'Query management' mock data (interim)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by Daniel-Lam — APPROVED** [review]
@johnbenjamin-hmcts - I've given the PR a quick once-over and haven't spotted anything that needs correction; approved

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    partyName: 'John Smith - Appellant',
+    roleOnCase: null,
+    partyMessages: [
+      {
+        id: 'case-message-001',
+        subject: 'Review attached document',
+        body: 'Please review attached document and advise if hearing should proceed?',
+        attachments: [],
+        isHearingRelated: true,
+        hearingDate: new Date(2023, 0, 10),
```
Comment (ronaldmansveld) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts:14]: Dates will come from the API in string format, not as a date object

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      {
+        id: 'case-message-004',
+        subject: 'Games',
+        body: 'In addition to my previous message, the mobile phones must be switched off inside the premises.',
+        attachments: [],
+        isHearingRelated: true,
+        hearingDate: new Date(2023, 0, 10),
+        createdOn: new Date(2023, 0, 4),
+        createdBy: '2222-2222-2222-2222',
+        parentId: 'case-message-002'
```
Comment (ronaldmansveld) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts:48]: You cannot have multiple responses to 1 query.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/domain/query-management.model.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export interface PartyMessage {
+  id: string;
+  subject?: string;
+  body: string;
+  attachments?: Document[];
+  isHearingRelated: boolean;
+  hearingDate?: Date;
+  createdOn: Date;
+  createdBy: string;
+  parentId?: string;
```
Comment (ronaldmansveld) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/domain/query-management.model.ts:12]: missing name field

File: projects/ccd-case-ui-toolkit/src/lib/shared/domain/definition/display-context-enum.model.ts [hunk 4]
```diff
@@ -6,5 +6,6 @@ export enum DisplayContextParameter {
 
 export enum DisplayContextCustomParameter {
   CaseFileView = 'CaseFileView',
-  LinkedCases = 'LinkedCases'
+  LinkedCases = 'LinkedCases',
+  QueryManagement = 'QueryManagement'
```
Comment (ronaldmansveld) [inline projects/ccd-case-ui-toolkit/src/lib/shared/domain/definition/display-context-enum.model.ts:10]: Why is this needed as display context?

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        body: 'Can I play games in my phone when my solicitor is talking?',
+        attachments: [],
+        isHearingRelated: true,
+        hearingDate: new Date(2023, 0, 10),
+        createdOn: new Date(2023, 0, 3),
+        createdBy: '1111-1111-1111-1111'
+      },
+      {
+        id: 'case-message-003',
+        subject: 'Games',
```
Comment (ronaldmansveld) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts:30]: Subject should only exist on top-level message (one without a parent)

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      {
+        id: 'case-message-001',
+        subject: 'Review attached document',
+        body: 'Please review attached document and advise if hearing should proceed?',
+        attachments: [],
+        isHearingRelated: true,
+        hearingDate: new Date(2023, 0, 10),
+        createdOn: new Date(2023, 0, 3),
+        createdBy: '1111-1111-1111-1111'
+      },
```
Comment (ronaldmansveld) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts:17]: missing name (see also comment on model)

**Review by ronaldmansveld — CHANGES_REQUESTED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/domain/definition/display-context-enum.model.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/domain/definition/display-context-enum.model.ts:10]: This DisplayContextCustomParameter is used in Palette Service's componentLauncherRegistry.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts]: I have changed the data type from date to string.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts:39]: I have now updated the mock data.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/domain/query-management.model.ts (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/domain/query-management.model.ts:13]: I have added the name field and also updated the mock data.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts (hunk 5 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts]: I have updated the mock data accordingly.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts (hunk 6 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/__mocks__/index.ts:18]: I have now updated the mock data and the model.

**Review by ronaldmansveld — APPROVED** [review]
